### PR TITLE
Azurik updates

### PR DESCRIPTION
**Pull request by blueshogun96**
These three fixes should be enough to get Azurik: Rize of Perathia working more acceptably (tested with NTSC version).

### CODE REVIEW

File: src/CxbxKrnl/EmuFile.cpp [hunk 1]
```diff
@@ -423,6 +433,19 @@ NTSTATUS CxbxConvertFilePath(
 				HostPath.append(1, '\\');
 		}
 
+		// Lastly, remove any '\\' sequences in the string (this should fix the problem with Azurik game saves)
+		replace_all( RelativePath, "\\\\", "\\" );
+
+		if (g_bPrintfOn) {
```
Comment (PatrickvL) [inline src/CxbxKrnl/EmuFile.cpp]: This looks like a duplicate of the code below?

**Comment by PatrickvL** [comment]
Nice work, and thanks for this PR!

I may have found one small issue - it's a remark on one file. Please review

**Comment by LukeUsher** [comment]
Sonic Heroes E3 Demo barely works at all anyway, it's not a big loss at this point. Fix what Patrick mentioned (the duplication) and we'll merge. Sonic Heroes E3 can be investigated as a separate incident.

File: src/CxbxKrnl/EmuFile.cpp [hunk 2]
```diff
@@ -331,6 +331,16 @@ void copy_string_to_PSTRING_to(std::string const & src, const xboxkrnl::PSTRING
 	memcpy(dest->Buffer, src.c_str(), dest->Length);
 }
 
+void replace_all(std::string& str, const std::string& from, const std::string& to) {
```
Comment (PatrickvL) [inline src/CxbxKrnl/EmuFile.cpp]: Credit this to https://stackoverflow.com/a/24315631 ?

File: src/CxbxKrnl/EmuFile.cpp (hunk 1 above)
Comment (blueshogun96) [reply src/CxbxKrnl/EmuFile.cpp]: Yeah, my mistake.  I'm fixing that now.

Shogun

**Comment by PatrickvL** [comment]
Thanks!

File: src/CxbxKrnl/EmuFile.cpp (hunk 2 above)
Comment (blueshogun96) [reply src/CxbxKrnl/EmuFile.cpp]: Okay, taking your advice on this one.  I was lazy and did a copy/paste from a different stackoverflow question.

Shogun

**Pull request merged**